solr/core/src/java/org/apache/solr/update/AddUpdateCommand.java

```diff
@@ -228,11 +225,25 @@ private void buildDocsList() {
     return unwrappedDocs;
   }
 
+  private static Collection<String> getChildDocumentsKeys(SolrInputDocument doc) {
```

Comment:
But this is only called in one place; no?  I think it's better if recUnwrapRelations does this logic itself as it avoids the intermediate HashSet.